### ISSUE TITLE
[Metric] It's not possible to format non-numeric value

### ISSUE THREAD

**Issue by mbondyra** [elastic/elastic-charts#2206]
**Describe the issue**

1. Go here: https://codesandbox.io/s/long-legend-sizing-forked-h83q6s?file=/src/App.tsx:127-134 and analyze the code, specifically the passed `valueFormatter`.
2. It will never run if the value is not a number
3. If you change it to number, the `valueFormatter` runs

Specifically, here's my usecase: I am working on [adding other types of data ](https://github.com/elastic/kibana/issues/137756)to be supported by visualization ('string', 'boolean', 'number', 'ip', 'date') to match the legacy metric. 

<img width="350" alt="Screenshot 2023-10-18 at 14 00 14" src="https://github.com/elastic/elastic-charts/assets/4283304/3e848aa1-62cc-42b9-adcc-9c1be46a3ea9">

While strings, booleans and ip don't need extra format to be passed, we need it for dates as we get them as strings. Not sure if there's another way to handle that. Let me know if that's not a bug and there is!

**Comment by mbondyra** [comment]
Just an update from my side: I found a way of passing already a formatted value. That's probably not ideal, but works! So instead of 
```
{
    value,
    valueFormatter: formatPrimaryMetric,
}
```

I pass 
```
{
    zvalue: formatPrimaryMetric(value),
}
```

**Comment by nickofthyme** [comment]
Good point @mbondyra I think in #1817 we just assumed it would be redundant to format a string but I think there is definitely a use for this. In any case I think it's strange the `valueFormatter` has no affect with string values.
